lib/mutant/require_highjack.rb

```diff
@@ -10,10 +10,14 @@ module RequireHighjack
     # @return [#call]
     #   the original implementation on singleton
     def self.call(target, callback)
-      target.method(:require).tap do
+      target.public_method(:require).tap do
         target.module_eval do
+          undef_method(:require)
           define_method(:require, &callback)
-          public :require
+          class << self
```

Comment:
> I did not assume this is public

I *think* it is. The Ruby documentation doesn't *usually* have any entries for private methods afaik.